### DISCUSSION TITLE
Grant User Access to Service EndPoint but not Page View

### DISCUSSION THREAD

**Discussion by flashyjunior** [serenity-is/Serenity#7172] [Help] [answered]
Please is it possible to allow a user to read a service endpoint but not access to the Page itself.
I realized when you give a user View access, the user automatically has access to the Page entity and the Service Endpoint as well.
I want to restrict users from accessing a page but still be able to lookup an entity in a lookupeditor

**Comment by luongnv6** [comment]
you can create a Custom Lookup, pls see this demo
https://github.com/serenity-is/common-features/blob/master/src/Serenity.Demo.Northwind/Modules/Customer/CustomerCityLookup.cs

**Comment by dfaruque** [comment, marked as answer]
Use `[NavigationPermission("MyModule:MyEntity:Navigation")]` on the Row.
Then, grant that the user only read permission, not navigation permission.
